Sync `webrtc-encoded-transform` (excluding tentative) from WPT upstream

Ahmad-S792 opened the pull request:
#### fd9700a1ac23c168ab0e4b9951b7ec6d3ba1a1a7
<pre>
Sync `webrtc-encoded-transform` (excluding tentative) from WPT upstream

<a href="https://bugs.webkit.org/show_bug.cgi?id=275529">https://bugs.webkit.org/show_bug.cgi?id=275529</a>

Reviewed by Youenn Fablet.

This patch is to sync tests as below:

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/ed9e9309618bdf76de06ff85757edbc8e1d7da82">https://github.com/web-platform-tests/wpt/commit/ed9e9309618bdf76de06ff85757edbc8e1d7da82</a>

* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/routines.js:
(async getNextMessage):
(async postMethod):
(async createWorker):
(async createTransform):
(async createTransforms):
(async createConnectionWithTransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-audio-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-change-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-late-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-metadata-transform-worker.js:
(onrtctransform.async waitForDetachAndPostMetadata):
(onrtctransform.process):
(onrtctransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-metadata-transform.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-metadata-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-worker.js:
(onrtctransform.process):
(onrtctransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-write-twice-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/sframe-keys.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/sframe-transform-in-worker.https.html:
* LayoutTests/tests-options.json:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/helper.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk-worker.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame-simulcast.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame-simulcast.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/w3c-import.log:
* LayoutTests/TestExpectations: Crashing Test added as Skip
* LayoutTests/platform/ios/TestExpectations: Platform Specific Expectation
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.https-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk.https-expected.txt: Ditto

Canonical link: <a href="https://commits.webkit.org/280174@main">https://commits.webkit.org/280174@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/736b84547e1b6eac4f08fef5fd413853788ec97f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55926 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35250 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8394 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58922 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6358 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42871 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6554 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45030 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4391 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57955 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33142 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48217 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26168 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29925 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5545 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4501 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/51894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5816 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60512 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5939 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52462 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48286 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51976 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12392 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31079 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32163 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33245 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31911 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf reviewed (approved):
Can you file a bug for the crashing test please?

Commit:
Sync `webrtc-encoded-transform` (excluding tentative) from WPT upstream

https://bugs.webkit.org/show_bug.cgi?id=275529

Reviewed by Youenn Fablet.

This patch is to sync tests as below:

Upstream commit: https://github.com/web-platform-tests/wpt/commit/ed9e9309618bdf76de06ff85757edbc8e1d7da82

* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/routines.js:
(async getNextMessage):
(async postMethod):
(async createWorker):
(async createTransform):
(async createTransforms):
(async createConnectionWithTransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-audio-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-change-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-late-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-metadata-transform-worker.js:
(onrtctransform.async waitForDetachAndPostMetadata):
(onrtctransform.process):
(onrtctransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-metadata-transform.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-metadata-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-worker.js:
(onrtctransform.process):
(onrtctransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-write-twice-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/sframe-keys.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/sframe-transform-in-worker.https.html:
* LayoutTests/tests-options.json:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/helper.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk-worker.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame-simulcast.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame-simulcast.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/w3c-import.log:
* LayoutTests/TestExpectations: Crashing Test added as Skip
* LayoutTests/platform/ios/TestExpectations: Platform Specific Expectation
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-generateKeyFrame.https-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-bad-chunk.https-expected.txt: Ditto

Canonical link: https://commits.webkit.org/280174@main
